src/components/Synthetics/MarketCard/MarketCard.tsx

```diff
@@ -34,24 +34,65 @@ export function MarketCard({ marketInfo, allowedSlippage, isLong, isIncrease }:
 
   const longShortText = isLong ? t`Long` : t`Short`;
 
-  const { liquidity, maxReservedUsd, reservedUsd, borrowingRate, fundingRate, totalInterestUsd, priceDecimals } =
-    useMemo(() => {
-      if (!marketInfo) return {};
-
-      return {
-        liquidity: getAvailableUsdLiquidityForPosition(marketInfo, isLong),
-        maxReservedUsd: getMaxReservedUsd(marketInfo, isLong),
-        reservedUsd: getReservedUsd(marketInfo, isLong),
-        borrowingRate: getBorrowingFactorPerPeriod(marketInfo, isLong, CHART_PERIODS["1h"]).mul(100),
-        fundingRate: getFundingFactorPerPeriod(marketInfo, isLong, CHART_PERIODS["1h"]).mul(100),
-        totalInterestUsd: marketInfo.longInterestUsd.add(marketInfo.shortInterestUsd),
-        priceDecimals: marketInfo.indexToken.priceDecimals,
-      };
-    }, [marketInfo, isLong]);
+  const {
+    liquidity,
+    maxReservedUsd,
+    reservedUsd,
+    borrowingRate,
+    fundingRateLong,
+    fundingRateShort,
+    totalInterestUsd,
+    priceDecimals,
+  } = useMemo(() => {
+    if (!marketInfo) return {};
 
+    return {
+      liquidity: getAvailableUsdLiquidityForPosition(marketInfo, isLong),
+      maxReservedUsd: getMaxReservedUsd(marketInfo, isLong),
+      reservedUsd: getReservedUsd(marketInfo, isLong),
+      borrowingRate: getBorrowingFactorPerPeriod(marketInfo, isLong, CHART_PERIODS["1h"]).mul(100),
+      fundingRateLong: getFundingFactorPerPeriod(marketInfo, true, CHART_PERIODS["1h"]).mul(100),
+      fundingRateShort: getFundingFactorPerPeriod(marketInfo, false, CHART_PERIODS["1h"]).mul(100),
+      totalInterestUsd: marketInfo.longInterestUsd.add(marketInfo.shortInterestUsd),
+      priceDecimals: marketInfo.indexToken.priceDecimals,
+    };
+  }, [marketInfo, isLong]);
+  const fundingRate = isLong ? fundingRateLong : fundingRateShort;
   const indexName = marketInfo && getMarketIndexName(marketInfo);
   const poolName = marketInfo && getMarketPoolName(marketInfo);
 
+  const fundingRateElements = useMemo(() => {
```

Comment:
I removed sign variables, as they are really redundant.
but I don't see a reason in a separate function as I'm already having a function which is also memoized

Reply:
Sorry, I think you misunderstood my point.

There is nothing wrong with the function `fundingRateElements`. To me this part of the code look a bit weird and not easy to read.

```
<Trans>{fundingRateElements[0]}</Trans>
<br />
<br />
<Trans>{fundingRateElements[1]}</Trans>
```

- We can do the logic of rendering long or short in the function itself instead of accessing the individual elements
- We don't need to use Trans again here
- And because the function contains the rendering logic we can name it starting with `render`. It's up to you to create one or two functions for long or short funding rate.